### PR TITLE
Filter overdue results by district

### PR DESCRIPTION
**Pull request by prabhanshuguptagit**
**Story card:** [ch4735](https://app.clubhouse.io/simpledotorg/story/4735/create-a-mobile-overdue-line-list-download)

## Because

Follow up to #2874. It is no longer feasible to show overdue results for `All facilities` on the dashboard.

## This addresses

This filters the overdue results by districts by default and then allows users to further filter by a facility.

https://user-images.githubusercontent.com/16774200/131478313-ef471a0d-0faa-4dbf-9159-0f6f3084083f.mov

### CODE REVIEW

File: app/controllers/concerns/overdue_list_filtering.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+    private
+
+    def set_accessible_facilities
+      @accessible_facilities = current_admin.accessible_facilities(:manage_overdue_list)
+    end
+
+    def populate_districts
+      @districts =
+        Region.district_regions.where(source_id: @accessible_facilities.map(&:facility_group_id).uniq).order(:name)
```
Comment (harimohanraj89) [inline app/controllers/concerns/overdue_list_filtering.rb:19]: Can we traverse this via facility regions instead? I'd like to minimize reliance on FGs and maximize reliance on regions, so it becomes easier to modify/do away with FGs in the future.

File: app/views/shared/_paginate_and_filter_by_facility.html.erb [hunk 2]
```diff
@@ -1,31 +1,40 @@
 <%= bootstrap_form_with(url: form_url, method: :get, layout: :horizontal, class: "mt-4") do |form| %>
-  <% form_id = local_assigns[:form_id]
-     facility_html_options = { form: form_id, onchange: "this.form.submit();", class: "selectpicker", data: {live_search: true} }.compact
-     page_html_options = { form: form_id, onchange: "this.form.submit();" }.compact
-  %>
+  <% html_select_options = { form: local_assigns[:form_id], onchange: "this.form.submit();" }.compact
+     searchable_select_options = html_select_options.merge(class: "selectpicker", data: {live_search: true}) %>
```
Comment (harimohanraj89) [inline app/views/shared/_paginate_and_filter_by_facility.html.erb:3]: Thanks for cleaning this up 😅

**Review by harimohanraj89 — APPROVED** [review]
Couple notes!

File: app/controllers/concerns/overdue_list_filtering.rb (hunk 1 above)
Comment (prabhanshuguptagit) [reply app/controllers/concerns/overdue_list_filtering.rb]: Addressed in 7f46c0a12

File: app/controllers/concerns/overdue_list_filtering.rb [hunk 3]
```diff
@@ -0,0 +1,39 @@
+module OverdueListFiltering
+  extend ActiveSupport::Concern
+
+  included do
+    before_action :set_accessible_facilities, only: [:index]
+    before_action :populate_districts, only: [:index]
+    before_action :set_selected_district, only: [:index]
+    before_action :populate_facilities, only: [:index]
+    before_action :set_selected_facility, only: [:index]
```
[the remark this answers is not in the document]
Comment (prabhanshuguptagit) [reply app/controllers/concerns/overdue_list_filtering.rb]: Addressed in a08318828

**Pull request merged**